modules/javafx.graphics/src/main/java/com/sun/javafx/scene/text/TextLayout.java

```diff
@@ -91,6 +93,28 @@ public Hit(int charIndex, int insertionIndex, boolean leading) {
         public int getCharIndex() { return charIndex; }
         public int getInsertionIndex() { return insertionIndex; }
         public boolean isLeading() { return leading; }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(charIndex, insertionIndex, leading);
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            if (this == obj)
+                return true;
+            if (obj == null)
+                return false;
+            if (getClass() != obj.getClass())
```

Comment:
any reason why this code uses getClass() != obj.getClass() ?

perhaps a better choice might be the usual pattern
```
if(x == this) {
 return true;
} else if(x instanceof Hit h) {
 return charIndex == h.charIndex && ...
}
return false;
```

Reply:
There are long discussions about this, and not really worth going into. They both have their place.  Generally, the `getClass` versions are easier to get right without violating the `equals` contract.  `instanceof` can be used, to allow comparisons with (trivial) subtypes, but then you must make `equals` final.  Anything fancier violates the equals contract.